src/decision/location_selection.jl

```diff
@@ -0,0 +1,296 @@
+        target_site_ids = collect(1:length(domain.site_ids))
+    end
+
+    for (scen_idx, scen) in enumerate(eachrow(scenarios))
+        depth_criteria = (domain.site_data.depth_med .<= (scen.depth_min .+ scen.depth_offset)) .& (domain.site_data.depth_med .>= scen.depth_min)
+        depth_priority = findall(depth_criteria)
+
+        considered_sites = target_site_ids[findall(in(depth_priority), target_site_ids)]
+        mcda_vars_temp = DMCDA_vars(domain, scen, considered_sites,  sum_cover[scen_idx, :], area_to_seed,
+            vec((mean(wave_scens[:, :, target_wave_scens], dims=(:timesteps, :scenarios)) .+ std(wave_scens[:, :, target_wave_scens], dims=(:timesteps, :scenarios))) .* 0.5),
+            vec((mean(dhw_scens[:, :, target_dhw_scens], dims=(:timesteps, :scenarios)) .+ std(dhw_scens[:, :, target_dhw_scens], dims=(:timesteps, :scenarios))) .* 0.5),
+            )
+
+        ranks_store(; scenarios=scen_idx, sites=considered_sites) .= _location_selection(
+            domain, mcda_vars_temp, scen.guided
+        )
+    end
+    # Set filtered locations as n_locs+1 for consistency with time dependent ranks
+    ranks_store[ranks_store.==0.0] .= length(domain.site_ids)+1
+
+    return ranks_store
+end
+function rank_locations(
+    domain::Domain,
+    scenarios::DataFrame,
+    sum_cover::NamedDimsArray,
+    area_to_seed::Float64,
+    agg_func::Function,
+    iv_type::Union{Int64, Symbol, String};
+    target_seed_sites=nothing,
+    target_shade_sites=nothing
+)::AbstractArray
+    ranks = rank_locations(
+        domain,
+        scenarios,
+        sum_cover,
+        area_to_seed;
+        target_seed_sites=target_seed_sites,
+        target_shade_sites=target_shade_sites,
+    )
+    local iv_id
+    try
+        iv_id = axiskeys(ranks, :intervention)[iv_type]
+    catch err
+        if !(err isa ArgumentError)
+            rethrow(err)
+        end
+
+        iv_id = iv_type
+    end
+
+    return agg_func(ranks(intervention=iv_id))
+end
+
+
+"""
+    ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)
+    ranks_to_frequencies(ranks::NamedDimsArray{D,T,3,A}; n_ranks=length(ranks.sites), agg_func=x -> dropdims(sum(x; dims=:timesteps); dims=:timesteps),) where {D,T,A}
+    ranks_to_frequencies(ranks::NamedDimsArray{D,T,2,A}; n_ranks=length(ranks.sites), agg_func=nothing) where {D,T,A}
+
+Post-process location ranks as given by `rank_locations()`.
+
+Returns the frequency with which each location was ranked across scenarios.
+
+Note: By default, ranks are aggregated over time where `ranks` is a 3-dimensional array.
+
+# Arguments
+- `ranks` : Location ranks from `rank_locations()`
+- `n_ranks` : Number of rankings (default is number of locations).
+- `agg_func` : Aggregation function to appy after frequencies are calculated.
+
+# Returns
+Frequency with which each location was selected for each rank.
+"""
+function ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)
```

Comment:
@Rosejoycrocker is `n_ranks` ever set by the user?

I can't think of a case where we'd want to set `n_ranks` less or more than the number of locations.

Reply:
My original thought with this is if you have a large number of sites (such as in the whole of GBR domain), you might only be interested in the frequencies of ranks 1:k where k is less than the total number of sites. This way you could save  time by not considering all the ranks. Not sure how relevant it is. I could set it as an optional parameter maybe?

Reply:
It already is an optional argument for the higher level functions. I was wondering if we could simplify the function signatures. I'm not sure if we need this flexibility but okay with leaving it in - might turn out to be useful.